ballerina/request_processor.bal

```diff
@@ -51,6 +51,29 @@ isolated function processSubscriptionVerification(http:Caller caller, http:Respo
     }
 }
 
+isolated function processUnsubscriptionVerification(http:Caller caller, http:Response response, 
+                                                    RequestQueryParams params, HttpToWebsubAdaptor adaptor) {
+    UnsubscriptionVerification message = {
+        hubMode: <string>params?.hubMode,
+        hubTopic: <string>params?.hubTopic,
+        hubChallenge: <string>params?.hubChallenge,
+        hubLeaseSeconds: params?.hubLeaseSeconds
+    };
+
+    UnsubscriptionVerificationSuccess|error result = adaptor.callOnUnsubscriptionVerificationMethod(message);
+    if result is UnsubscriptionVerificationError {
+        response.statusCode = http:STATUS_NOT_FOUND;
+        var errorDetails = result.detail();
+        updateResponseBody(response, errorDetails["body"], errorDetails["headers"], result.message());
+    } else if result is error {
+        response.statusCode = http:STATUS_NOT_FOUND;
+        updateResponseBody(response, (), (), result.message());
```

Comment:
Instead of passing `nil` can't we make the argument optional ?

Reply:
As for the function declaration of [updateResponseBody](https://github.com/ballerina-platform/module-ballerina-websub/blob/082afecc652e54fe76ef23361201163de07274bf/ballerina/utils.bal#L205) I do not think we could mark these parameters optional. According to the usage of `updateResponseBody`, `reason` parameter is the one which will be optional most the time.